components/Frame/Header.js

```diff
@@ -253,6 +283,25 @@ class Header extends Component {
               </a>
             </div>
           )}
+          {opaque && (
+            <button
+              {...styles.search}
+              role='button'
+              title={t('header/nav/search/aria')}
+              onClick={(e) => {
+                e.preventDefault()
+                e.stopPropagation()
+                if (isSearchActive) {
+                  window.scrollTo(0, 0)
+                } else {
+                  Router.pushRoute('search').then(() => window.scrollTo(0, 0))
```

Comment:
Gave it a quick try and couldn't really get your suggestion working. Since I can't access the URL/path of the previous history entry before actually executing Router.back() or similar, how would I be able to implement the default origin? My guess is I'd need some app level state keeping a reference to lastPage or so.

Reply:
You'd think that the Next router would provide access to the current history (at least through something like [`history.lenght`](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/history.md)).  I agree, we probably need to track the history ourselves, or at least have something like a `hasNavigationOccurred` flag.

Reply:
You can use a global variable in the header (or wherever you manage the icon).

See `beforeSearch` global being set in [`onSearch`](https://github.com/lobbywatch/website/blob/05ab20f96425b021c97ce6e7dba3044fb5524929/src/components/Frame/Header.js#L189) of lobbywatch.ch. The behaviour there is a bit more complex because it searches as you type and goes back by clearing the field.